pkg/interop/native/ledger/transaction_signer.go

```diff
@@ -0,0 +1,65 @@
+package ledger
+
+import "github.com/nspcc-dev/neo-go/pkg/interop"
+
+// TransactionSigner represent the signer of a NEO transaction. It's similar to
+// Signer class in Neo .net framework.
+type TransactionSigner struct {
+	// Bytes represents the binary serialized form of the given signer.
+	Bytes []byte
+	// Account represents the account (160 bit BE value in a 20 byte slice) of
+	// the given signer.
+	Account interop.Hash160
+	// Scopes represents the set of witness scopes of the given signer.
+	Scopes SignerScope
+	// Contracts represents the set of contract hashes (160 bit BE value in a 20
+	// byte slice) allowed to be called by the signer.
+	AllowedContracts []interop.Hash160
+	// AllowedGroups represents the set of contract groups (ecdsa public key
+	// bytes in a 32 byte slice) allowed to be called by the signer.
+	AllowedGroups []interop.PublicKey
+	// Rules represents a rule-based witness scope of the given signer.
+	Rules []WitnessRule
+}
+
+// SignerScope represents a signer's witness scope.
+type SignerScope byte
+
+// Various witness scopes.
+const (
+	// None specifies that no contract was witnessed. Only sign the transaction.
+	None SignerScope = 0
+	// CalledByEntry means that this condition must hold:
+	// EntryScriptHash == CallingScriptHash. No parameters are needed, as the
+	// witness/permission/signature given on first invocation will automatically
+	// expire if entering deeper internal invokes. This can be default safe
+	// choice for native NEO/GAS.
+	CalledByEntry SignerScope = 0x01
+	// CustomContracts define custom hash for contract-specific witness.
+	CustomContracts SignerScope = 0x10
+	// CustomGroups define custom public key for group members.
+	CustomGroups SignerScope = 0x20
+	// Rules is a set of conditions with boolean operators.
+	Rules SignerScope = 0x40
+	// Global allows this witness in all contexts. This cannot be combined with
+	// other flags.
+	Global SignerScope = 0x80
+)
+
+// WitnessRule represents a single rule for Rules witness scope.
+type WitnessRule struct {
+	Action    WitnessAction
+	Condition interface{} // TODO: impossible to convert?
```

Comment:
There has to be some explanation, or helper types that this `Condition` can be cast to according to its type.

Reply:
Fixed, but the difficulty is that we have two different stackitem structures for casting. The first one has tha condition value, and the second on (CalledByEntry) has only one element (the type). Added a note about it.